cm_ovirt_vm_creator.py

```diff
@@ -172,13 +173,26 @@ def create_vms(cluster_nodes, args):
                 time.sleep(args.sleep_between_iterations)
                 vm = vm_service.get()
                 print("vm.status = %s, vm.fqdn= '%s'" % (vm.status, vm.fqdn), file=sys.stderr)
-                if vm.status == types.VmStatus.UP or counter > 20:
+                if vm.status == types.VmStatus.UP:
                     break
+                counter += 1
+
             if vm.status != types.VmStatus.UP:
                 print("ERROR - VM {0} still not up after 20 retries".format(node), file=sys.stderr)
                 sys.exit(-1)
             else:
-                vm_dict[node] = find_vm_ip(vm_service)
+                ip = None
+                counter = 1
+                while counter < args.num_of_iterations:
+                    ip = find_vm_ip(vm_service)
+                    if ip is not None:
+                        break
+                    counter += 1
+                    msg = "waiting for ip... {0}/{1} attempts".format(counter, args.num_of_iterations)
+                    print(msg, file=sys.stderr)
+                    time.sleep(args.sleep_between_iterations)
+
+                vm_dict[node] = ip
```

Comment:
We wait for the machine to be up and then to get an IP. 
Do we really need to check if it is up or does the second check suffice?

If you do need to run two wait loops it would do good to extract a function that waits iterations and evaluates a block.

Reply:
Yes, we have to wait for the IP to be there, because sometimes the machine comes up but the guest agent is slow to start/respond so the IP is not yet available to the API.

> extract a function that waits iterations and evaluates a block.

Anonymous functions are not possible in Python as far as I know, so I'd need to extract every single step (wait for vm to be created, wait for vm to be up, wait for ip) into its own function, and I think that's a bit of an overkill, especially since I'm going to re-write this function soon to start the VMs asynchronously instead of waiting for each VM to start before starting the next one.

Reply:
> Yes, we have to wait for the IP to be there, because sometimes the machine comes up but the guest agent is slow to start/respond so the IP is not yet available to the API.

I meant only doing the IP check (and if that fails due to the machine not being up go to next cycle) so we have only one loop.

Reply:
I'm theoretically fine with that (we'd just need to raise the default number of attempts, because I'm not sure 20 will be sufficient with just one loop).

@bazulay what do you think?

Reply:
I suggest doing the IP loop at the end (after finished to create the VMs and let them come up) simply because it will take too much time to wait for each IP (till it is reported by the guest agent) .